src/components/Tooltip.js

```diff
@@ -0,0 +1,135 @@
+    text: PropTypes.string.isRequired,
+
+    // Child to wrap with Tooltip. Must be only child.
+    children: PropTypes.oneOfType([
+        PropTypes.func,
+        PropTypes.element,
+        PropTypes.instanceOf(Component),
+    ]).isRequired,
+};
+
+class Tooltip extends Component {
+    constructor(props) {
+        super(props);
+
+        this.animation = new Animated.Value(0);
+
+        // The wrapper view containing the wrapped content along with the Tooltip itself.
+        this.wrapperView = null;
+        this.width = 0;
+        this.height = 0;
+
+        // The tooltip (popover) itself.
+        this.tooltip = null;
+        this.tooltipWidth = 0;
+        this.tooltipHeight = 0;
+
+        // The distance between the left side of the rendered view and the left side of the window
+        this.xOffset = 0;
+
+        // The distance between the top of the rendered view and the top of the window
+        this.yOffset = 0;
+
+        this.getPosition = this.getPosition.bind(this);
+        this.showTooltip = this.showTooltip.bind(this);
+        this.hideTooltip = this.hideTooltip.bind(this);
+    }
+
+    /**
+     * Get the position and measure the size of the wrapper view and the tooltip itself.
+     */
+    getPosition() {
+        this.wrapperView.measureInWindow((x, y, width, height) => {
+            this.xOffset = x;
+            this.yOffset = y;
+            this.width = width;
+            this.height = height;
+        });
+        this.tooltip.measureInWindow((x, y, width, height) => {
+            this.tooltipWidth = width;
+            this.tooltipHeight = height;
+        });
+    }
+
+    /**
+     * Display the tooltip in an animation.
+     */
+    showTooltip() {
+        Animated.timing(this.animation, {
+            toValue: 1,
+            duration: 140,
+        }).start();
+    }
+
+    /**
+     * Hide the tooltip in an animation.
+     */
+    hideTooltip() {
+        Animated.timing(this.animation, {
+            toValue: 0,
+            duration: 140,
+        }).start();
+    }
+
+    render() {
+        const interpolatedSize = this.animation.interpolate({
+            inputRange: [0, 1],
+            outputRange: [0, 1],
+        });
+        const {
+            animationStyle,
+            tooltipWrapperStyle,
+            tooltipTextStyle,
+            pointerWrapperStyle,
+            pointerStyle,
+        } = getTooltipStyles(
+            interpolatedSize,
+            this.xOffset,
+            this.yOffset,
+            this.width,
+            this.height,
+            this.tooltipWidth,
+            this.tooltipHeight,
+        );
+
+        return (
+            <Hoverable
+                onHoverIn={this.showTooltip}
+                onHoverOut={this.hideTooltip}
+            >
+                <View
+                    ref={el => this.wrapperView = el}
+                    onLayout={this.getPosition}
+                    collapsable={false}
```

Comment:
What will happen if this is `collapsable={true}`?

Reply:
I'm not 100% sure if it's completely necessary, but from the docs:

> Views that are only used to layout their children or otherwise don't draw anything may be automatically removed from the native hierarchy as an optimization. Set [collapsable] to false to disable this optimization and ensure that this View exists in the native view hierarchy.

I was just trying to make sure that `this.wrapperView.measureInWindow` doesn't error because this wrapper view doesn't render anything that's not in another view.

Reply:
Gooot it. Sorta sounds like the docs are saying that if this has no visible children it will be removed from the view hierarchy. But this makes sense to me since if this `View` had no visible children then it should also not need a `Tooltip` either? 🤔  

Anyways, I'm still not sure what we are preventing. Maybe we can add this when it becomes a solution to a problem we've run into? Unless you've already observed that problem then we should add a comment.

Reply:
Yeah, my concern was that the tooltip itself would not always be visible, so it would be removed from the native hierarchy and not measured before it needs to be shown. I'll try remove it and see if it causes any problems